fusor-ember-cli/app/components/text-f.js

```diff
@@ -56,25 +65,24 @@ export default Ember.Component.extend({
     'invalidIsAlphaNumeric',
     'invalidIsHostname',
     'invalidNetworkRange',
+    'invalidCIDRNotation',
     function() {
       return (this.get('showValidationError') &&
-              (Ember.isPresent(this.get('errors.name')) ||
+               (Ember.isPresent(this.get('errors.name')) ||
                this.get('doesntMatchPassword') ||
                this.get('passwordTooShort') ||
                this.get('validIsRequiredAndBlank') ||
                this.get('validIsUnique') ||
                this.get('invalidIsAlphaNumeric') ||
                this.get('invalidIsHostname') ||
-               this.get('invalidNetworkRange')
-              )
+               this.get('invalidNetworkRange') ||
+               this.get('invalidCIDRNotation'))
```

Comment:
+1 @cfchase, I deliberately patched this the way things have been done, but you're right, we need a way to address this strategically as it grows. I like the idea of keeping it out of the controllers as well. 

@isratrade, FYI, I integrated the [ember-validations](https://github.com/dockyard/ember-validations) addon in pursuit of this, but it was pretty cumbersome and I think Chris correctly argued it didn't really simplify things. I still have the patch, but I don't think it's the right tool for the job and we're probably better off implementing something ourselves a la Chris' suggestion.